### ISSUE TITLE
Image Alt & Title text not processed with htmlspecialchars()

### ISSUE THREAD

**Issue by HartLarsson** [PrestaShop/PrestaShop#19022]
#### Describe the bug

By inspecting the image alt, if the alt text is containing a double quote will broke the alt parameter.

#### Expected behavior

every alt text must be processed with htmlspecialchars() before write into HTML

#### Steps to Reproduce

1. Open a new product
2. Ad an image and write in Caption field something like: text "textinside" text final
3. Save and preview the product page
4. Inspect the product image alt text is the same as title. It should be "textinside"

The same problem there is for **title** tag.

#### Additional information

* PrestaShop version: 1.7.6.5
* PHP version: 7.2.30

**Comment by florine2623** [comment]
Hi @HartLarsson ,
Thanks for the recommandation. I'll let our product team know about it!
Thanks!

**Comment by florine2623** [comment]
Hi @PrestaShop/prestashop-product-team ,
Do you think using `htmlspecialchars()` is relevant in alt image of product ?
Thanks !

**Comment by HartLarsson** [comment]
> Hi @PrestaShop/prestashop-product-team ,
> Do you think using `htmlspecialchars()` is relevant in alt image of product ?
> Thanks !

absolutely yes, if you not fix the alt and title field from special chars you get problem on validation.

for example could be used a code like that:

`{if !empty($product.cover.legend)}
alt="{$product.cover.legend|escape:'htmlall'}" title="{$product.cover.legend|escape:'htmlall'}"
{else}
alt="{$product.name|escape:'htmlall'}"
{/if}`

or 
`{if !empty($product.cover.legend)}
alt="{$product.cover.legend|htmlspecialchars'}" title="{$product.cover.legend|htmlspecialchars}"
{else}
alt="{$product.name|htmlspecialchars}"
{/if}`

**Comment by LouiseBonnard** [comment]
Dear @HartLarsson, thanks for this issue! @PrestaShop/prestashop-core-developers, what do you think?

**Comment by davidglezz** [comment]
I have not been able to reproduce the problem. By default the text is escaped, the problem occurs if we put nofilter: `title="{$image.legend nofilter}"`

Although the browser shows that there is a double quotation mark in the middle, it is because it has already decoded it, but the source code is valid.

![image](https://user-images.githubusercontent.com/194784/87727133-aee34680-c7c0-11ea-9c21-545dc6830810.png)

![image](https://user-images.githubusercontent.com/194784/87727207-d0443280-c7c0-11ea-92d8-8b82a0586102.png)

Tested with today's develop 1.7.8 version

I think we can close the issue

**Comment by colinegin** [comment]
ping @florine2623 did you reproduce this issue ?

**Comment by PierreRambaud** [comment]
Hi everyone,

I don't manage to reproduce the error on the front, but I have an error on the back office :sweat_smile: 
![image](https://user-images.githubusercontent.com/1462701/96017692-97ad0700-0e4a-11eb-9c79-f01ca8218bde.png)

**Comment by ghost** [comment]
@PierreRambaud you have an error in the bacloffice because  you add > or < and the validate of legend is `isGenericName`

If we change by `isCleanHTML` it's ok and we can see ( encoded ) < or > in the alt image

I can made a PR if it's ok for you

**Comment by hibatallahAouadni** [comment]
Hello @HartLarsson 

With the New Product Page V2, I didn't reproduce the issue but I notice that the saved caption is not displayed in alt, see the attached screen record below:

[issue_19022.webm](https://github.com/PrestaShop/PrestaShop/assets/23308427/1eae8530-4830-42ef-95fa-0367e2ed472c)

Did you reproduce the same issue? if so, could you please change the content and the title of issue :pray: 
Waiting for your feedback.

Thanks!

**Comment by florine2623** [comment]
I have reproduced the same issue as @hibatallahAouadni in PS `8.1.x` and `develop`.

The alt text is configured differently than the alt text. Here it is the same.

<img width="863" alt="Screenshot 2023-10-12 at 17 16 36" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/4fdbb4cb-9341-41aa-af10-38ad8a4bcccc">

<img width="1383" alt="Screenshot 2023-10-12 at 17 16 20" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/3a30476b-1e90-4acd-a135-e5e2d1883c07">

**Comment by HartLarsson** [comment]
> I have reproduced the same issue as @hibatallahAouadni in PS `8.1.x` and `develop`.
> 
> The alt text is configured differently than the alt text. Here it is the same.
> 
> <img alt="Screenshot 2023-10-12 at 17 16 36" width="863" src="https://user-images.githubusercontent.com/16019289/274647212-4fdbb4cb-9341-41aa-af10-38ad8a4bcccc.png"> <img alt="Screenshot 2023-10-12 at 17 16 20" width="1383" src="https://user-images.githubusercontent.com/16019289/274647229-3a30476b-1e90-4acd-a135-e5e2d1883c07.png">

i noticed too on 8.1.x :(

**Comment by StoreCommanderDev** [comment]
Hello

I noticed this error with legend too on 8.1.x.

It seems it has been resolved into 8.1.3
look at this commit
https://github.com/PrestaShop/PrestaShop/commit/4644883c5e38073f13d5c2edbe202d8a63fec5ee

or this PR: https://github.com/PrestaShop/PrestaShop/pull/34689

**Comment by MatShir** [comment]
I did the test. @StoreCommanderDev, it seems to be fixed, yes ! Thanks for the heads-up
